site/en/docs/extensions/mv3/service_workers/service-worker-lifecycle/index.md

```diff
@@ -23,7 +23,8 @@ Next is the extension's [`onInstalled`](/docs/extensions/reference/runtime/#even
 event to set a state or for one-time initialization, such as a [context menu](/docs/extensions/reference/contextMenus/).
 
 ```js
-chrome.runtime.onInstalled.addListener(() => {
+chrome.runtime.onInstalled.addListener((details) => {
+  if(details.reason !== "install") return;
```

Comment:
Would you be ok with this? Context menus are cleared on update so we need to add them back there too.

```suggestion
  if(details.reason !== "install" && details.reason !== "update") return;
```